andela/balder-ah-backend
# #163519791 Un-bookmarking an article should be done via the article id for ease

#### What does this PR do?

This PR modifies the bookmark controller to allow users un-bookmark an article via the article id

#### Description of Task to be completed?
 - Modify bookmark controller to allow users unbookmark an article via article id
 - Modify middleware to check for existence of article via the article id


#### How should this be manually tested?
- Clone this repo
- In your terminal cd into the folder it creates
- Run **npm install** to install all the dependencies that will be used by the project
- Check the **.env.sample** to create a **.env** file in your root folder
- Run **npm run dev:start** to start the app
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Send a GET request to **localhost:3000/api/user/bookmarks** to retrieve all the user's bookmarks 
- Send a DELETE request to **localhost:3000/api/user/bookmarks/:articleId** to remove a particular bookmark
- Run **npm test** to test the app
- View the Swagger UI on **localhost:3000/api/docs**


#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories? (If applicable)

[#163519791](https://www.pivotaltracker.com/story/show/163519791)

#### Screenshots (If applicable)

N/A

#### Questions:
